python/ray/experimental/raysort/main.py

```diff
@@ -190,146 +208,270 @@ def _dummy_sort_and_partition(part: np.ndarray,
 
 
 @ray.remote
-def mapper(boundaries: List[int], mapper_id: PartId,
-           path: Path) -> List[ray.ObjectRef]:
+@tracing_utils.timeit("map")
+def mapper(args: Args, boundaries: List[int], path: Path) -> List[np.ndarray]:
     logging_utils.init()
-    task_id = f"M-{mapper_id} Mapper"
-    logging.info(f"{task_id} starting {args}")
+
     if args.skip_input:
-        block_size = int(np.ceil(args.input_part_size / args.num_reducers))
-        return [
-            ray.put(
-                np.frombuffer(np.random.bytes(block_size), dtype=np.uint8))
-            for _ in range(args.num_reducers)
-        ]
+        part = np.frombuffer(
+            np.random.bytes(args.input_part_size), dtype=np.uint8)
+    else:
+        part = _load_partition(path)
 
-    part = _load_partition(path)
     sort_fn = _dummy_sort_and_partition \
         if args.skip_sorting else sortlib.sort_and_partition
     blocks = sort_fn(part, boundaries)
-    logging.info(f"{task_id} saving to object store")
-    return [ray.put(part[offset:offset + size]) for offset, size in blocks]
+    return [part[offset:offset + size] for offset, size in blocks]
 
 
-def _dummy_merge(blocks: List[np.ndarray], _n: int) -> Iterable[memoryview]:
-    for block in blocks:
+def _dummy_merge(
+        num_blocks: int, _n: int,
+        get_block: Callable[[int, int], np.ndarray]) -> Iterable[np.ndarray]:
+    blocks = [((i, 0), get_block(i, 0)) for i in range(num_blocks)]
+    while len(blocks) > 0:
```

Comment:
Can you explain what the dummy_merge does? I found this a bit hard to follow.

Reply:
It simulates merging `n` blocks. `get_block(i, d)` returns the `d`-th chunk of block `i`. This function repeatedly calls `get_block` until it exhausts all the chunks of all the blocks.